src/com/google/enterprise/adaptor/UrlValidator.java

```diff
@@ -0,0 +1,71 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.enterprise.adaptor;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Validates URLs by syntax checking and validating the host is reachable.
+ * No attempts to fetch the URL are made, so a mocked up URL with
+ * placeholder values in format substitutions may be used.
+ */
+public class UrlValidator {
+  /** The logger for this class. */
+  private static final Logger log =
+      Logger.getLogger(UrlValidator.class.getName());
+
+  /** The connect timeout, in milliseconds. */
+  private static final int TIMEOUT = 30 * 1000;
+
+  /**
+   * Attempts to validate the given URL syntax and host reachability.
+   * In this case, we're mostly trying to catch typos.
+   *
+   * @param urlString the URL to test
+   * @return {@code true} if the URL's host is reachable, {@code false}
+   *         if the host is not reachable.
+   * @throws MalformedURLException if the URL can not be parsed
+   */
+  public boolean validate(String urlString) throws MalformedURLException {
+    URL url = new URL(urlString);
```

Comment:
I usually see URI used in this role. Not sure if there's a compelling argument either way. URI handles escaping better, which may impact the host lookup.